src/include/storage/store/table.h

```diff
@@ -43,26 +42,21 @@ struct KUZU_API TableScanState {
     TableScanState(common::ValueVector* nodeIDVector,
         std::vector<common::ValueVector*> outputVectors,
         std::shared_ptr<common::DataChunkState> outChunkState)
-        : tableID{common::INVALID_TABLE_ID}, nodeIDVector(nodeIDVector),
-          outputVectors{std::move(outputVectors)}, outState{outChunkState.get()},
-          semiMask{nullptr} {
+        : table{nullptr}, nodeIDVector(nodeIDVector), outputVectors{std::move(outputVectors)},
+          outState{std::move(outChunkState)}, semiMask{nullptr} {
         rowIdxVector = std::make_unique<common::ValueVector>(common::LogicalType::INT64());
-        rowIdxVector->state = std::move(outChunkState);
+        rowIdxVector->state = outState;
     }
 
-    TableScanState(common::table_id_t tableID, std::vector<common::column_id_t> columnIDs,
-        std::vector<const Column*> columns = {},
-        std::vector<ColumnPredicateSet> columnPredicateSets = {})
-        : tableID{tableID}, nodeIDVector(nullptr), outState{nullptr},
-          columnIDs{std::move(columnIDs)}, semiMask{nullptr}, columns{std::move(columns)},
-          columnPredicateSets{std::move(columnPredicateSets)} {
-        rowIdxVector = std::make_unique<common::ValueVector>(common::LogicalType::INT64());
-    }
+    TableScanState(std::vector<common::column_id_t> columnIDs, std::vector<const Column*> columns)
```

Comment:
You might wanna double check if we need `columnIDs` & `columns` at the same time

Reply:
This is indeed redundant semantically, but as I'm not passing in `Table`, so they are necessary for now. But I think I should refactor `Column` to keep its own `columnID`, which makes more sense.